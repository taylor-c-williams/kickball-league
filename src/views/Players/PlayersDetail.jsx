import { useEffect, useState } from 'react';
import { Link, useParams, useHistory } from 'react-router-dom';
import { getPlayerById, deletePlayerById } from '../../services/players';

export default function PlayersDetail() {
	const { id } = useParams();
	const [player, setPlayer] = useState(null);
	const [loading, setLoading] = useState(true);
	const history = useHistory();

	useEffect(() => {
		getPlayerById(id)
			.then((res) => setPlayer(res))
			.finally(() => setLoading(false));
	}, [id]);

	if (loading) return <span className="loading">Loading...</span>;

	const handleDelete = async ({ id, name }) => {
		await deletePlayerById(id);
		history.push(`/teams/`);
	};

	return (
		<div className="content">
			<h1>{player.name}</h1>
			<h2>{player.position}</h2>
			<img
				width="300px"
				src="https://m.media-amazon.com/images/I/51vPSKT9L7L._AC_SX425_.jpg"
				alt="an adorable frog faced kickyball"
			/>
			<ul>
				<li>
					<Link to={`/teams/${player.teams.id}`}>{player.teams.name}</Link>
				</li>
				<li>
					{player.teams.city}, {player.teams.state}
				</li>
			</ul>
			<Link to={`/editPlayer/${player.id}`}>Update Player</Link>
			<button
				type="button"
				aria-label={`Delete ${player.name}`}
				onClick={() => handleDelete({ id: player.id, name: player.name })}
			>
				Delete Player
			</button>
		</div>
	);
}
