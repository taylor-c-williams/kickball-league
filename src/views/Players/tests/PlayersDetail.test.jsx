import {
	waitForElementToBeRemoved,
	render,
	screen,
} from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import PlayersDetail from '../PlayersDetail';

it('should render a detailed view of an individual player', async () => {
	const player = {
		created_at: '2021-12-10T01:44:26+00:00',
		id: 3,
		name: 'player name',
		position: 'player position',
		team_id: 1,
	};
	const detail = render(
		<MemoryRouter initialEntries={['/players/3']}>
			<Route path="/players/:id">
				<PlayersDetail player={player} />
			</Route>
		</MemoryRouter>
	);

	await waitForElementToBeRemoved(() => screen.queryByText(/Loading.../i));

	const update = await screen.findByText(/update/i);

	expect(detail).toMatchSnapshot();
	expect(update).toBeInTheDocument();
});
