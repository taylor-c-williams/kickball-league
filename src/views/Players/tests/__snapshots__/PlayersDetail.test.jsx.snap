// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a detailed view of an individual player 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content"
      >
        <h1>
          player name
        </h1>
        <h2>
          player position
        </h2>
        <img
          alt="an adorable frog faced kickyball"
          src="https://m.media-amazon.com/images/I/51vPSKT9L7L._AC_SX425_.jpg"
          width="300px"
        />
        <ul>
          <li>
            <a
              href="/teams/1"
            >
              team name
            </a>
          </li>
          <li>
            team city
            , 
            team state
          </li>
        </ul>
        <a
          href="/editPlayer/3"
        >
          Update Player
        </a>
        <button
          aria-label="Delete player name"
          type="button"
        >
          Delete Player
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content"
    >
      <h1>
        player name
      </h1>
      <h2>
        player position
      </h2>
      <img
        alt="an adorable frog faced kickyball"
        src="https://m.media-amazon.com/images/I/51vPSKT9L7L._AC_SX425_.jpg"
        width="300px"
      />
      <ul>
        <li>
          <a
            href="/teams/1"
          >
            team name
          </a>
        </li>
        <li>
          team city
          , 
          team state
        </li>
      </ul>
      <a
        href="/editPlayer/3"
      >
        Update Player
      </a>
      <button
        aria-label="Delete player name"
        type="button"
      >
        Delete Player
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
